app/tut/ga/scoip.md

```diff
@@ -0,0 +1,392 @@
+déanann sé seic an bhfuil `10` níos mó ná `1`, ar ndóigh tá sé, dá bhrí sin glaonn sé ar
+`scríobh_uimhir(9)`{.setanta}. Scríobhann é sin "9" agus glaonn sé ar `scríobh_uimhir(8)`{.setanta},
+ansin glaonn sé ar `scríobh_uimhir(7)`{.setanta}, `scríobh_uimhir(6)`{.setanta} srl. go dtí go
+sroicheann sé `scríobh_uimhir(1)`{.setanta}. Ag an bpointe sin níl `x > 1`{.setanta} a thuilleadh
+agus stopann sé.
+
+Má chruthaítear athróg sa ghníomh, cruthaítear [[leagan ar leith|a unique version]] gach uair a
+ghlaoitear ar an gníomh. [[Go sonrach|Specifically]], nuair a ghlaonn gníomh air é féin,
+[[ní roinneann na glaonna a n-athróga|the calls don't shared their variables]]. Féach ar an
+ríomhchlár seo:
+
+{{{
+gníomh f(x) {
+    y := x
+
+    má x > 1 {
+        f(x - 1)
+    }
+
+    scríobh(y)
+}
+
+f(2)
+}}}
+
+Nuair a ghlaoimid ar `f(2)`{.setanta}, cruthaíonn sé athróg nua `y` atá cothrom le `2`{.setanta}.
+Ansin déanann sé seic an bhfuil `x > 1`{.setanta}, agus tá sé. Ansin glaonn sé ar `f(1)`{.setanta}.
+Cruthaítear athróg nua anois `y` atá cothrom le `1`{.setanta}, **tá an athróg seo difriúil leis an
+athróg `y` a chruthaíomar cheana.** Níl `1`{.setanta} níos mó ná `1`{.setanta}, mar sin ní
+ghlaoitear ar `f(0)`{.setanta}. Ansin scríobhann sé `y` ar an gconsól, mar sin scríobhann sé "1".
+Ansin téann sé ar ais go dtí an gníomh roimhe `f(2)`{.setanta} agus leanann sé ar aghaidh. Is é
+`scríobh(y)`{.setanta} an gcéad líne eile, ach **níl `y` cothrom le 1 anois, mar tá an athróg `y`
+difriúil leis an `y` a scríobhamar cheana.** Tá an `y` seo cothrom le `2`{.setanta}, dá bhrí sin
+scríobhann sé "2" ar an gconsól.
+
+# An Scóip Uilíoch
+
+Tá scóip speisialta i *Setanta*, an [[**scóip uilíoch**|global scope]]. Tá scóip uilíoch ag athróg
+atá sainmhínithe ag an [[barrleibhéal|top level]]: sa bhloc ag barr an ríomhchlár. Mar shampla, tá
+scóip uilíoch ag `x`, `mo_ghníomh` agus `MoChreatlach` sa ríomhchlár a leanas, ach níl scóip uilíoch
+ag an athróg `áitiúil`.
+
+```{.setanta .numberLines}
+creatlach MoChreatlach {
+    gníomh ainm() {
+        scríobh("Setanta")
+    }
+}
+
+gníomh mo_ghníomh() {
+    áitiúil := "Dia duit"
+    toradh áitiúil
+}
+
+x := 3
+```
+
+Tá athróga uilíocha speisialta, mar is féidir leat iad a úsáid **sula gcruthaíonn tú iad** nuair a
+úsáideann tú iad i gníomh éigin. Féachaimis ar cúpla sampla:
+
+Oibríonn an cód seo: Bain triail as!
+
+{{{
+gníomh abair_mo_ainm() {
+    scríobh("Dia duit", ainm)
+}
+
+ainm := "Eoin"
+
+abair_mo_ainm()
+}}}
+
+Ní oibríonn cód mar sin ach amháin sa scóip uilíoch, mar shampla, ní oibríonn cód mar seo: Má
+ritheann tú é, gheobhaidh tú earráid.
+
+{{{
+le i idir (0, 10) {
+    gníomh abair_mo_ainm() {
+        scríobh("Dia duit", ainm)
+    }
+
+    ainm := "Eoin"
+
+    abair_mo_ainm()
+}
+}}}
+
+Is féidir linn é seo a úsáid chun dhá ghníomh a cruthú mar seo:
+
+```{.setanta .numberLines}
+gníomh uimhir_ré(x) {
```

Comment:
Tá an péire gníomh seo beagán ait - is dócha go n-oibríonn siad, ach b'fhiú iad a mhíniú!

Reply:
Curtha isteach, GRMA